From 96f849cfd2aa1d40309ba050a79e9070a69af01b Mon Sep 17 00:00:00 2001
From: Jon DeVree <nuxi@vault24.org>
Date: Mon, 19 Sep 2016 21:02:55 -0400
Subject: [PATCH 1/2] Add udev_ctrl_cleanup() to manager_exit()

This fixes the initial bug added in 693d371 (udevd: move main-loop to sd-event)
which results in manager_exit() deleting the udev_ctrl structure before
calling the cleanup routine on it.

Signed-off-by: Jon DeVree <nuxi@vault24.org>
---
 src/udev/udevd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/udev/udevd.c b/src/udev/udevd.c
index 19f1c29..1236f73 100644
--- a/src/udev/udevd.c
+++ b/src/udev/udevd.c
@@ -741,6 +741,8 @@ static void manager_exit(Manager *manager) {
                   "STOPPING=1\n"
                   "STATUS=Starting shutdown...");
 
+        udev_ctrl_cleanup(manager->ctrl);
+
         /* close sources of new events and discard buffered events */
         manager->ctrl_event = sd_event_source_unref(manager->ctrl_event);
         manager->ctrl = udev_ctrl_unref(manager->ctrl);
-- 
2.9.3

